mne_bids/path.py

```diff
@@ -120,22 +132,88 @@ def __str__(self):
                     key = 'rec'
                 basename.append('%s-%s' % (key, val))
 
-        if self.suffix is not None:
-            basename.append(self.suffix)
+        if self.kind is not None:
+            if self.extension is not None:
+                basename.append(f'{self.kind}{self.extension}')
+            else:
+                basename.append(self.kind)
 
         basename = '_'.join(basename)
-        if self.prefix is not None:
-            basename = op.join(self.prefix, basename)
-
         return basename
 
+    @property
+    def fpath(self):
+        """Get the full filepath for this BIDS file.
```

Comment:
Drop the `Get`, as it's not an ordinary function anymore.
```suggestion
        """The full filepath for this BIDS file.
```
I'm not sure what a "full filepath" is supposed to mean. Full path, perhaps?

Reply:
So we can't represent a full path because we can't represent `sub-01/ses-02/eeg/`, but we can represent `sub-01/ses-02/eeg/sub-01_ses-02_run-01_eeg.fif`.

Lmk thoughts on this. I'm not entirely convinced this is the way, but if one wants to represent a BIDS folder path, then they would ideally use `make_bids_folders`.

Reply:
My point was that my understanding of the term "path" seems to be more generic than yours. To you, a path seems to be a location of a **directory**, whereas for me, it's the location of **any** object on the filesystem, be it a directory or a file – a URI.

Quoting Wikipedia:

> A path, the general form of the name of a file or directory, specifies a unique location in a file system.

https://en.wikipedia.org/wiki/Path_(computing)

Reply:
Okay that makes sense. So as it operates right now... `BIDSPath` can't operate as a general path object because it can't represent directories easily. It can only represent explicit files within the BIDS root. Because we only represent file paths, it allows us to infer the file path from given entities using globbing. 

I suppose if we want to represent directory paths as well, we would need some different abstraction and thinking to allow the BIDSPath object to know when it is getting a directory vs when it is getting a file path. 

How do you think we should proceed?